docs/contributing_to_vhive.md

```diff
@@ -0,0 +1,82 @@
+﻿# Contributing to vHive
+This document gives a brief overview of how you can contribute to the vHive repository.
+
+## Pull Requests
+When contributing to the repository you should work in a seperate branch and create a GitHub pull request for your branch. For all pull requests to vHive we require that you do the following:
+- Sync your Repo
+- Squash commits
+- Rebase on Main
+- Avoid Merges
+
+### Syncing your GitHub Repos
+When you are working on a fork of the vHive repository, keeping your fork in sync with the main repository keeps your workspace up-to-date and reduces the risk of merge conflicts. 
+
+The most common way of syncing up your fork is with a remote that points to the upstream repository:
+1. If you have not done so already, create a new remote for the upstream vHive repo:
+	```bash
+	git remote add upstream https://github.com/ease-lab/vhive.git
+	```
+	You can always check your existing remotes with `git remote -v`.
+2. Fetch branches and commits from the upstream (vHive) repo:
+	```bash
+	git fetch upstream
+	```
+3. Switch to your local default branch (named `main` by default):
+	```bash
+	git checkout main
+	```
+4. Merge the upstream changes:
+	```bash
+	git merge upstream/main
+	```
+
+You can check out [the official GitHub docs](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/syncing-a-fork) for more information on syncing forks.
+
+### Squashing Commits
+We prefer for every commit in the repo to encapsulate a single concrete and atomic change/addition. This means our commit history shows a clear and structured progression, and that it remains consise and readable. In your own branch you can clean up your commit history by squashing smaller commits together using `git rebase`:
+1. Find the hash of the oldest commit which you want to squash up to. For example, if you made three commits in sequence (A, B, C, such that C is the latest commit) and you wanted to squash A and B together, you would need to find the hash of A. You can find the hash of a commit on GitHub or by using the command:

```

Comment:
For three commits A < B < C in time, if I want to squash A and B together, I need the hash of "A-1". The correct example would be squashing B and C (and finding the hash of A). Indeed reading further, this is I believe what you meant.

Reply:
thanks for the correction, resolved.